inc/ticket.class.php

```diff
@@ -31,811 +31,856 @@
+        global $DB;
+
+       //get old ticket
+        $ticket = new Ticket();
+        if (!$ticket->getFromDB($tickets_id)) {
+            Session::addMessageAfterRedirect(__('Error : get old ticket', 'escalade'), false, ERROR);
+            exit;
+        }
+
+       //set fields
+        $fields = $ticket->fields;
+        $fields = array_map(['Toolbox', 'addslashes_deep'], $fields);
+        $fields['id']                  = 0;
+        $fields['_users_id_requester'] = 0;
+        $fields['status']              = CommonITILObject::INCOMING;
+
+       /*var_dump($fields);
+       exit;*/
+
+       //create new ticket (duplicate from previous)
+        if (! $newID = $ticket->add($fields)) {
+            Session::addMessageAfterRedirect(__('Error : adding new ticket', 'escalade'), false, ERROR);
+            exit;
+        }
+
+       //add link between them
+        $ticket_ticket = new Ticket_Ticket();
+        if (
+            !$ticket_ticket->add([
+                'tickets_id_1' => $tickets_id,
+                'tickets_id_2' => $newID,
+                'link'         => Ticket_Ticket::LINK_TO
+            ])
+        ) {
+            Session::addMessageAfterRedirect(__('Error : adding link between the two tickets', 'escalade'), false, ERROR);
+            exit;
+        }
+
+       //add a followup to indicate duplication
+        $followup = new ITILFollowup();
+        if (
+            ! $followup->add([
+                'items_id'        => $newID,
+                'itemtype'        => Ticket::class,
+                'users_id'        => Session::getLoginUserID(),
+                'content'         => __("This ticket has been cloned from the ticket num", "escalade") . " " .
                               $tickets_id,
-         'is_private'      => true,
-         'requesttypes_id' => 6 //other
-      ])) {
-         Session::addMessageAfterRedirect(__('Error : adding followups', 'escalade'),false, ERROR);
-         exit;
-      }
-
-      //add actors to the new ticket (without assign)
-      //users
-      $query_users = "INSERT INTO glpi_tickets_users
+                'is_private'      => true,
+                'requesttypes_id' => 6 //other
+            ])
+        ) {
+            Session::addMessageAfterRedirect(__('Error : adding followups', 'escalade'), false, ERROR);
+            exit;
+        }
+
+       //add actors to the new ticket (without assign)
+       //users
+        $query_users = "INSERT INTO glpi_tickets_users
       SELECT null AS id, $newID as tickets_id, users_id, type, use_notification, alternative_email
       FROM glpi_tickets_users
       WHERE tickets_id = $tickets_id AND type != 2";
-      if (!$res = $DB->query($query_users)) {
-         Session::addMessageAfterRedirect(__('Error : adding actors (user)', 'escalade'),false, ERROR);
-         exit;
-      }
-      //groups
-      $query_groups = "INSERT INTO glpi_groups_tickets
+        if (!$res = $DB->doQuery($query_users)) {
```

Comment:
It look like `$tickets` is sent from the client, which mean this could become an SQL injection on GLPI 11 when autoescaping is removed.

We should either keep the deprecated `query` call or not use a raw string query here.